src/binding/petsc4py/src/petsc4py/PETSc/MatPartitioning.pyx

```diff
@@ -23,38 +24,134 @@ cdef class MatPartitioning(Object):
     def __call__(self):
         return self.getValue()
 
-    def view(self, Viewer viewer=None):
+    def view(self, Viewer viewer=None) -> None:
+        """View the partitioning data structure.
+
+        Collective.
+
+        Parameters
+        ----------
+        viewer
+            A `Viewer` to display the graph.
+
+        See also
+        --------
+        petsc.MatPartitioningView
+
+        """
         assert self.obj != NULL
         cdef PetscViewer vwr = NULL
         if viewer is not None: vwr = viewer.vwr
         CHKERR( MatPartitioningView(self.part, vwr) )
 
-    def destroy(self):
+    def destroy(self) -> Self:
+        """Destroy the partitioning context.
+
+        Collective.
+
+        See also
+        --------
+        petsc.MatPartitioningDestroy, create
+
+        """
         CHKERR( MatPartitioningDestroy(&self.part) )
         return self
 
-    def create(self, comm=None):
+    def create(self, comm: Comm | None = None) -> Self:
+        """Create a partitioning context.
+
+        Collective.
+
+        Parameters
+        ----------
+        comm
+            The MPI communicator.
```

Comment:
or None for the defaut comm .....

Reply:
We have to do a final pass for these too